### PR TITLE
makeTemporaryToolkitFolder always creates folders within a toolkit-wide temp folder

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Description

The toolkit had a regression that caused it to create all temp folders at the same level as the toolkit's temp folder, instead of within it. This change fixes the behavior.

## Related Issue(s)

#474

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/filesystemUtilities.ts [hunk 1]
```diff
@@ -73,7 +73,11 @@ export async function findFileInParentPaths(searchFolder: string, fileToFind: st
 
 const mkdtemp = promisify(fs.mkdtemp)
 export const makeTemporaryToolkitFolder = async (...relativePathParts: string[]) => {
-    const _relativePathParts = relativePathParts || ['vsctk']
+    const _relativePathParts = relativePathParts || []
+    if (_relativePathParts.length === 0) {
+        _relativePathParts.push('vsctk')
+    }
+
     const tmpPath = path.join(tempDirPath, ..._relativePathParts)
```
Comment (kiiadi) [inline src/shared/filesystemUtilities.ts]: where does `tempDirPath` come from - if this is system wide you might run into a problem that the JetBrains toolkit faced : https://github.com/aws/aws-toolkit-jetbrains/issues/950

sam has a limitation that the 'template' and source folders must be relative within the same drive-mount. On windows if you create a temporary template on `C:\` for example but the source workspace is on `D:\` then it doesn't work.

You should update the temporary folder to use something relative to the workspace. I'm not sure if VS Code exposes this ability.

Comment (bryceitoc9) [reply src/shared/filesystemUtilities.ts]: https://github.com/aws/aws-toolkit-vscode/blob/9a9082736f3f665711083a793eaa8b2454216ba0/src/shared/filesystemUtilities.ts#L16

It's only used for tests and uses the Node `os.tmpdir` value.

Comment (awschristou) [reply src/shared/filesystemUtilities.ts]: Let me verify this. We've never tried running with a second drive. I'll update the comments here and add an issue if there is one.

Comment (awschristou) [reply src/shared/filesystemUtilities.ts]: @kiiadi  tested by making a SAM App on a D drive (windows and temp are on C drive), init, run (performs a build), debug (performs a build) were successful.

**Pull request merged**